# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/DNSPacket.h
# Commit: af0f14eed70e
# Full Hash: af0f14eed70edb2abb1ecd3cae4928387287d3e1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 9f913cbd60ee -r af0f14eed70e netwerk/dns/DNSPacket.h
--- a/netwerk/dns/DNSPacket.h	Thu Jun 13 15:08:04 2024 +0200
+++ b/netwerk/dns/DNSPacket.h	Fri Jun 14 09:11:49 2024 +0000
@@ -52,6 +52,8 @@
   nsresult OnDataAvailable(nsIRequest* aRequest, nsIInputStream* aInputStream,
                            uint64_t aOffset, const uint32_t aCount);
 
+  // Encode the input host name into a sequence of labels.
+  static nsresult EncodeHost(nsCString& aBody, const nsACString& aHost);
   // Encodes the name request into a buffer that represents a DNS packet
   virtual nsresult EncodeRequest(nsCString& aBody, const nsACString& aHost,
                                  uint16_t aType, bool aDisableECS);